ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RemoteHostExecutor.java

```diff
@@ -68,28 +66,14 @@ private RemoteDataAugmenter(
     this.ksqlClient = Objects.requireNonNull(ksqlClient);
   }
 
-  public static RemoteDataAugmenter create(
-      final String statementText,
+  public static RemoteHostExecutor create(
+      final ConfiguredStatement<? extends StatementWithExtendedClause> statement,
       final SessionProperties sessionProperties,
       final KsqlExecutionContext executionContext,
       final SimpleKsqlClient ksqlClient
   ) {
-    return new RemoteDataAugmenter(
-        statementText, sessionProperties, executionContext, ksqlClient);
-  }
-
-
-  public <R> R augmentWithRemote(
-      final R localResult,
-      final BiFunction<R, Pair<List<KsqlEntity>, Set<HostInfo>>, R> mergeFunc
-  ) {
-    Objects.requireNonNull(mergeFunc);
-    if (sessionProperties.getInternalRequest()) {
-      return mergeFunc.apply(localResult, new Pair<>(ImmutableList.of(), ImmutableSet.of()));
-    }
-    final Pair<List<KsqlEntity>, Set<HostInfo>> remoteResults = fetchAllRemoteResults();
-
-    return mergeFunc.apply(localResult, remoteResults);
+    return new RemoteHostExecutor(
+        statement.getStatementText(), sessionProperties, executionContext, ksqlClient);
```

Comment:
Similar to other comment about `statement` vs `statementText`. If we aren't using the statement could we cut out the `create` method and just use the constructor directly?

Reply:
The reason for this is a bit subtle, but I don't know if I understood the semantics of ConfiguredStatement correctly. The idea is to let people know that it only runs one clause e.g. `describe streams;` instead of multiple e.g. `describe streams; describe queries;`

If I allowed arbitrary strings I would have to support arbitrary arity of ksql sentences and that spun out of control a bit too quickly

Reply:
Another reason for this was so that only statements that support `extended` clause can be passed into this executor

Reply:
Ahh, that makes sense as a method of control, thanks for clarifying